contentpal/src/test/java/org/dmfs/android/contentpal/tables/SyncedTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.tables;
+
+import android.net.Uri;
+import android.provider.ContactsContract;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.InsertOperation;
+import org.dmfs.android.contentpal.Operation;
+import org.dmfs.android.contentpal.Predicate;
+import org.dmfs.android.contentpal.Table;
+import org.dmfs.android.contentpal.UriParams;
+import org.dmfs.android.contentpal.tools.uriparams.EmptyUriParams;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+
+/**
+ * @author Marten Gajda
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = "src/test/AndroidManifest.xml")
+public class SyncedTest
+{
+    @Test
+    public void testInsertOperation() throws Exception
+    {
+        final InsertOperation<Object> resultOperation = mock(InsertOperation.class);
+        Table<Object> testTable = new Synced<>(new AbstractMockTable()
```

Comment:
Not sure if that was better but I looked it up how this could be handled with Mockito instead of this custom `AbstractMockTable`.
There is this that `ArgumentCaptor` but as they say here:
https://static.javadoc.io/org.mockito/mockito-core/2.2.9/org/mockito/ArgumentCaptor.html
it can make tests harder to read,
however there is this `ArgumentMather` which could be better:
https://static.javadoc.io/org.mockito/mockito-core/2.2.9/org/mockito/ArgumentMatcher.html
(mockito should be updated to 2.1 for this new interface)

Just mentioning, not suggesting that it should be changed now.